plugins/channel-safely/channel-groups.cpp

```diff
@@ -0,0 +1,272 @@
+#include <channel-groups.h>
+#include <tile-cache.h>
+#include <inlines.h>
+#include <modules/Maps.h>
+#include <df/block_square_event_designation_priorityst.h>
+
+#include <random>
+
+
+
+// adds map_pos to a group if an adjacent one exists, or creates one if none exist... if multiple exist they're merged into the first found
+void ChannelGroups::add(const df::coord &map_pos) {
+    // if we've already added this, we don't need to do it again
+    if (groups_map.count(map_pos)) {
+        return;
+    }
+    /* We need to add map_pos to an existing group if possible...
+     * So what we do is we look at neighbours to see if they belong to one or more existing groups
+     * If there is more than one group, we'll be merging them
+     */
+    df::coord neighbors[8];
+    get_neighbours(map_pos, neighbors);
+    Group* group = nullptr;
+    int group_index = -1;
+
+    DEBUG(groups).print("    add(" COORD ")\n", COORDARGS(map_pos));
+    // and so we begin iterating the neighbours
+    for (auto &neighbour: neighbors) {
+        // go to the next neighbour if this one doesn't have a group
+        if (!groups_map.count(neighbour)) {
+            TRACE(groups).print(" -> neighbour is not designated\n");
+            continue;
+        }
+        // get the group, since at least one exists... then merge any additional into that one
+        if (!group){
+            TRACE(groups).print(" -> group* has no valid state yet\n");
```

Comment:
I see no problem

Reply:
I think it would only be a problem if you needed to get trace info from a Windows user who was hitting a bug. I don't think NDEBUG is ever defined when building on Linux

Reply:
sounds like a problem for another PR

Reply:
agreed, I didn't mean to imply that changes to Debug.h belong in this PR.